src/django_components/templatetags/component_tags.py

```diff
@@ -81,55 +65,34 @@ def _get_components_from_preload_str(preload_str: str) -> List["Component"]:
     return components
 
 
-@register.simple_tag(name="component_dependencies")
-def component_dependencies(preload: str = "") -> SafeString:
+def _component_dependencies(kind: Literal["js", "css", "both"]) -> SafeString:
```

Comment:
Refactored the logic for the `{% component_dependencies %}`  tags so that the logic is defined only once.